Automattic/jetpack
# Social | Extract connection toggle state into a hook.

Currently the logic for connection toggles on post edit page is quite complicated and unmanageable. This PR attempts to untangle the code and fix a resulting issue.

## Proposed changes:

*  Create `useConnectionState` hook to extract the connection state logic
* Fix the misleading connection sharing count

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to post edit page
* Open Jetpack or Social sidebar
* Confirm that connection state works same as before
* Confirm that the message `This post will be shared to N connections.` works as expected.
